tpls/mdspan/include/experimental/__p2642_bits/layout_padded.hpp

```diff
@@ -59,6 +59,9 @@ MDSPAN_INLINE_FUNCTION constexpr size_t get_actual_static_padding_value() {
   } else {
     return dynamic_extent;
   }
+#ifdef __INTEL_COMPILER
+  __builtin_unreachable();
+#endif
```

Comment:
I would rather use the same workaround as for `nvcc` directly below which is `return 0;`.

Reply:
Yeah just use the same as NVCC

Reply:
Thanks for the change in https://github.com/kokkos/mdspan/pull/349

Reply:
I mean we still need to pull the changes from https://github.com/kokkos/mdspan/pull/349 in to `Kokkos` manually.

Reply:
Yes, understood 👍

Reply:
I'm assuming they will come through a snapshot update, but let me know if I should apply them as part of this PR

Reply:
I think it would make sense to pull them in through this pull request. Otherwise, this won't fix the issue.

Reply:
I peeled and applied your patch from https://github.com/kokkos/mdspan/pull/349 , should be ready to go now